### ISSUE TITLE
[consistent-type-imports] support decorator metaData without specifying project

### ISSUE THREAD

**Issue by xboy2012** [typescript-eslint/typescript-eslint#4638]
For projects that use "emitDecoratorMetaData: true" in `tsconfig.json`

If we want to use `consistent-type-imports`, we have to specify `project` in `parserOptions`, in order to prevent the rule from reporting false positives.

Everything works OK as we configure as above. But we encounter performance issues. Linting time is too long.

As we specified `project` in `parserOptions`, then we uses type-aware linting(In other words, we need to compile source code by typescript). 

However, is it really the best practice to do like this?

+ I only need the parser to understand `emitDecoratorMetaData: true`
+ I do not need the time-costing type-aware linting.

Is there a way to just tell `consistent-type-imports` to act as if `emitDecoratorMetaData: true` is set, without really compiling with typescript?

such as add an option to the rule

```
"@typescript-eslint/consistent-type-imports": ['error', { emitDecoratorMetaData: true }]
```

**Comment by bradzacher** [comment]
> I do not need the time-costing type-aware linting

Have you checked out our list of type-aware lint rules? There are a lot of very valuable rules which can help catch many bugs!

----

This is a known weakness. As mentioned in https://github.com/typescript-eslint/typescript-eslint/issues/4608, we should add an option here.

**Comment by xboy2012** [comment]
> > I do not need the time-costing type-aware linting
> 
> Have you checked out our list of type-aware lint rules? There are a lot of very valuable rules which can help catch many bugs!
> 
> This is a known weakness. As mentioned in #4608, we should add an option here.

I know it. But just in this case, type-aware lint is not necessary. We can obviously tell the parser to act as if `emitDecoratorMetaData: true`, without having to specify `project`, which includes the unwanted  type-aware lint.

**Comment by xboy2012** [comment]
I think this is exactly what I want, as you said.

https://github.com/typescript-eslint/typescript-eslint/issues/4608#issuecomment-1054599223
